### DISCUSSION TITLE
Ctrl + Shift + S = SumatraPDF 'not responding'

### DISCUSSION THREAD

**Discussion by hrrrps** [sumatrapdfreader/sumatrapdf#4124] [Q&A]
Having trouble saving a PDF. It has just saved, but it took ages.

Windows 11, SumatraPDF v3.5.2 64 bit

**Comment by GitHubRulesOK** [comment]
time to save depends on a few factors 

1) save to non local locations will take many more times than local drive due to the nature of PDF construction and transport overheads
(It can take me an age to download from say internet archive.)
Silly analogy but explains why.
If my Server was on Mars and PDF only sent at the "Speed of Light" (cannot be faster). It takes roughly 22.5 minutes to start saving there then minutes to refactor (recompress) then save and another 22.5 minutes to get confirmation it went correctly but any spec of dust in that transmission beam it could add another 45 minutes to resend a missing part.

2) There is a difference in saving an annotated file rather than a simple duplicate. The main body will be sent untouched but the additions may be needing the whole file scanned several times for confirmation they were sealed correctly as the new length.

3) time taken when sending files is very much dependant on size and complexity of content as many systems need to decompress files in transport then recompress at the store.

4) Antivirus / Defender UAC Explorer and "Search" security monitors will need to authorise the EDITED file as safe to save thus it takes time for them to expand analyse and accept the file is benign enough to start saving (The more complex the file , the longer the time).

 For **me** on my device I have a problem saving downloads quickly in explorer, as some of the the above add a very large delay when simply accessing the downloads folder to enter the filename..

**Comment by kjk** [comment]
How big was the PDF?

Was it saved to a hard drive or network drive (like DropBox, OneDrive etc.)?

How long exactly is "ages"?  You can find that out from logs: `Ctrl + K` for command palette, `show logs`. There should be a line telling how long it took (`Saved annotations...` - I assume the context here is saving annotations).

But ultimately there's only so much we can do if the PDF is large and / or is saved to a slow / networked drive.

**Reply by hrrrps** [reply]
It is OneDrive using a Cryptomator container. PDF is 94MB. Ages is over a minute, but here are the log line:

Saved annotations to [document]' in  127314.91 ms, incremental: 1

**Reply by GitHubRulesOK** [reply]
<strike>13</strike> my mistake its 127  seconds  does seem a long time but that is the problems (as described above at least 45 minute overhead if its a server on Mars).

It is not the time of saving, it is the time of system transport to and fro possibly multiple times (say again, say again) with verification and other interference .

The potential for corruption due to comms brownout while editing ALWAYS suggests to be safe a file for edit should be pulled local edited and returned (Server File systems like SharePoint, use file systems of check out & check in)

Shared folders are to be avoided except for reading or final copy archive into.

**Reply by kjk** [reply]
How long does it take to save to your hard drive?

How long does it take to copy 94 MB file from hard drive to OneDrive?

127 seconds is slow but the large size of the file combined with the slow drive might just be inevitable
